src/renderer/screens/onboarding/steps/GenuineCheck/index.js

```diff
@@ -76,12 +81,18 @@ const GenuineCheck = (props: StepProps) => {
     setGenuineCheckModalOpened(false);
   }, []);
 
-  const handleGenuineCheckPass = useCallback(() => {
-    updateGenuineCheck({
-      isDeviceGenuine: true,
-    });
-    setGenuineCheckModalOpened(false);
-  }, [updateGenuineCheck]);
+  const handleGenuineCheckPass = useCallback(
+    ({ result }: any) => {
+      if (!hasCompletedOnboarding && (!result || !result.installed.length)) {
+        dispatch(setHasInstalledApps(false));
```

Comment:
we might need to discuss more, but i wonder if we simply shouldn't **at any point in time** just track the "last seen app installed" and use that as a fact so if user ever re-uninstall all apps and he don't have accounts, we would correctly invite him to go back to Manager (and not add accounts, because we remembered you had no apps)

Reply:
will change or not once we all agree on a flow on monday